NitroxClient/MonoBehaviours/IntroCinematicUpdater.cs

```diff
@@ -0,0 +1,164 @@
+
+        remotePlayerCustomHead = Instantiate(new GameObject("diveSuit_custom_head_geo"), Partner.PlayerModel.transform.Find("export_skeleton/head_rig"));
+        remotePlayerCustomHead.transform.localPosition = new Vector3(1.0941f, -0.1163f, -1.2107f);
+        remotePlayerCustomHead.transform.localRotation = new Quaternion(-0.05750692f, -0.5272675f, -0.6740523f, -0.5141357f);
+
+        MeshFilter mesh = remotePlayerCustomHead.AddComponent<MeshFilter>();
+        MeshRenderer meshRenderer = remotePlayerCustomHead.AddComponent<MeshRenderer>();
+        mesh.mesh = remotePlayerHeadRenderer.sharedMesh;
+        meshRenderer.materials = remotePlayerHeadRenderer.materials;
+        remotePlayerCustomHead.SetActive(false);
+
+        bool isLeftPlayer = NitroxServiceLocator.Cache<LocalPlayer>.Value.PlayerId < Partner.PlayerId;
+        introEndTarget = EscapePod.main.transform.Find("EscapePodCinematics/intro_end");
+        Vector3 introEndDiff = isLeftPlayer ? new Vector3(0, 0, 0.3f) : new Vector3(0, 0, -0.3f);
+        introRemoteEndPosition = introEndTarget.position - introEndDiff;
+        introEndTarget.position += introEndDiff;
+    }
+
+    private Vector3 remotePlayerDestinationPos;
+    private Quaternion remotePlayerDestinationRot;
+
+    private bool shouldDisplaceLocalPlayer;
+    private float localPlayerDisplacementTime;
+
+    private bool shouldDisplaceRemotePlayer;
+    private Vector3 remotePlayerDisplacement = Vector3.zero;
+
+    private IEnumerator Start()
+    {
+        remoteRenders.ForEach(smr => smr.enabled = false);
+        seatBarRendererRight.ForEach(smr => smr.updateWhenOffscreen = true);
+
+        yield return new WaitForSecondsRealtime(11f); //Local player in seat
+        remoteRenders.ForEach(smr => smr.enabled = true);
+
+        yield return new WaitForSecondsRealtime(33.45f - 11f); //Remote Player deadly hit => disable original head
+        remotePlayerHeadRenderer.enabled = false;
+
+        yield return new WaitForSecondsRealtime(33.58f - 33.45f); //Remote Player visible => enable custom head
+        remotePlayerCustomHead.SetActive(true);
+
+        yield return new WaitForSecondsRealtime(40f - 33.58f); //Seatbelt appears => enable position sync & switch to original head
+        remotePlayerCustomHead.SetActive(false);
+        remotePlayerHeadRenderer.enabled = true;
+        Destroy(remotePlayerCustomHead);
+
+        yield return new WaitForSecondsRealtime(50.5f - 40f); //Slowly move remote player backwards
+        shouldDisplaceRemotePlayer = true;
+
+        yield return new WaitForSecondsRealtime(50.72f - 50.5f); //Disable and move remote player
+        shouldDisplaceRemotePlayer = false;
+        remotePlayerDisplacement = Vector3.zero;
+        remoteRenders.ForEach(smr => smr.enabled = false);
+
+        yield return new WaitForSecondsRealtime(53f - 50.72f); //Slowly move local player to modified cin-end
+        shouldDisplaceLocalPlayer = true;
```

Comment:
Could these values be moved into constants? Seems like some repeated values that would also benefit from being named constants.

Reply:
The idea behind this is that the "time since cinematic started" is always at the left side of the subtraction. That way if we need to change the timing or insert something, we can just bring up a running stopwatch on the screen and know exactly when something is going to happen. I think making them to dedicated consts would just end up in "firstSegmentTime, secondSegmentTime".